src/pybind/mgr/prometheus/module.py

```diff
@@ -115,6 +116,174 @@ def health_status_to_number(status: str) -> int:
     alert_metric('SLOW_OPS', 'OSD or Monitor requests taking a long time to process'),
 ]
 
+HEALTHCHECK_DETAIL = ['name', 'severity']
+
+
+class HealthCheckEvent:
+    def __init__(self, name: str, severity: str, first_seen:float, last_seen: float, count: int, active:bool = True):
+        self.name = name
+        self.severity = severity
+        self.first_seen = first_seen
+        self.last_seen = last_seen
+        self.count = count
+        self.active = active
+
+    def as_dict(self):
+        return {
+            "name": self.name,
+            "severity": self.severity,
+            "first_seen": self.first_seen,
+            "last_seen": self.last_seen,
+            "count": self.count,
+            "active": self.active
+        }
```

Comment:
```suggestion
class HealthCheckEvent(NamedTuple):
        name: str
        severity: SeverityEnum
        first_seen: datetime
        last_seen: datetime
        count: int
        active: bool

HealthCheckEvent()._asdict()  # built-in method
```

Reply:
I Like! Thanks, I'll update

Reply:
Tried namedtuple but these are updatable objects, and namedtuple is immutable. Once we move to 3.7, I'd use datalass decorator